synapse/storage/databases/main/devices.py

```diff
@@ -1441,6 +1441,13 @@ def __init__(
             self._remove_duplicate_outbound_pokes,
         )
 
+        self.db_pool.updates.register_background_index_update(
+            "device_lists_changes_in_room_by_room_index",
+            index_name="device_lists_changes_in_room_by_room_idx",
+            table="device_lists_changes_in_room",
+            columns=["room_id", "stream_id"],
```

Comment:
Was https://github.com/matrix-org/synapse/pull/14516 some prerequisite work for this change (or at least to see a performance difference with the combination)?

Reply:
I think it was a combination really, but hard to know. The main aim of #14516 was to make it so that the rows of `device_lists_changes_in_room` on disk were roughly ordered by stream ID, making it much more efficient to look at rows in stream ID order. 

This PR mainly speeds up queries that were being done during sync, around "has anything changed since X in a particular set of rooms R"